src/mongoose_admin_api/mongoose_admin_api_domain.erl

```diff
@@ -0,0 +1,150 @@
+-module(mongoose_admin_api_domain).
+-behaviour(cowboy_rest).
+
+-export([init/2,
+         is_authorized/2,
+         content_types_provided/2,
+         content_types_accepted/2,
+         allowed_methods/2,
+         to_json/2,
+         from_json/2,
+         delete_resource/2]).
+
+-ignore_xref([to_json/2, from_json/2]).
+
+-import(mongoose_admin_api, [parse_body/1, try_handle_request/3, throw_error/2, resource_created/4]).
+
+-type req() :: cowboy_req:req().
+-type state() :: map().
+
+-spec init(req(), state()) -> {cowboy_rest, req(), state()}.
+init(Req, Opts) ->
+    mongoose_admin_api:init(Req, Opts).
+
+-spec is_authorized(req(), state()) -> {true | {false, iodata()}, req(), state()}.
+is_authorized(Req, State) ->
+    mongoose_admin_api:is_authorized(Req, State).
+
+-spec content_types_provided(req(), state()) ->
+          {[{{binary(), binary(), '*'}, atom()}], req(), state()}.
+content_types_provided(Req, State) ->
+    {[
+      {{<<"application">>, <<"json">>, '*'}, to_json}
+     ], Req, State}.
+
+-spec content_types_accepted(req(), state()) ->
+          {[{{binary(), binary(), '*'}, atom()}], req(), state()}.
+content_types_accepted(Req, State) ->
+    {[
+      {{<<"application">>, <<"json">>, '*'}, from_json}
```

Comment:
Wouldn't it be better to write it in one line?

Reply:
It is like this in all API modules, and I think it is more readable this way (multiple nested tuples).

Reply:
ok